tables/DataMan/Adios2StMan.cc

```diff
@@ -144,23 +147,23 @@ rownr_t Adios2StMan::getNrRows()
 //
 // impl class implementation using ADIOS2
 //
+
 Adios2StMan::impl::impl(Adios2StMan &parent, MPI_Comm mpiComm)
     : parent(parent)
 {
     itsMpiComm = mpiComm;
-    std::string engineType;
-    adios2::Params engineParams;
-    std::vector<adios2::Params> transportParams;
-    Adios2StMan(mpiComm, engineType, engineParams, transportParams);
+    Adios2StMan(mpiComm, "", {}, {}, {});
```

Comment:
The old version has the same problem, but I just saw it... should this be a delegating constructor in the member initialiser list, or am I misreading something? Otherwise we're just creating an `Adios2StMan` and quickly disposing of it, and the only thing that gets set is `itsMpiComm`. I guess this particular flavour of the constructor isn't used (much), otherwise we would have stepped into problems earlier.

In other words, I suspect this should be something like:

```c++
Adios2StMan::impl::impl(Adios2StMan &parent, MPI_Comm mpiComm)
  : Adios2StMan::impl(parent, mpiComm, "", {}, {}, {})
```

Alternatively we could rework the constructor list to have a single one with most of the arguments defaulted.

Reply:
@rtobar I was thinking about having a single constructor too. I thought this was used somewhere so you haven't removed. If I remember correctly, this has nothing special besides that it can quickly instantiate an Adios2StMan with all default configurations without going through the logics in the other constructor. But those logics are not expensive anyway, so I guess I will just remove it.

Reply:
This constructor is removed. Now there is only one constructor of Adios2StMan and Adios2StMan::impl, with default parameters in place.